File: commons/src/main/java/se/tink/android/redirection/RedirectionReceiver.java
package se.tink.android.redirection;

import java.util.List;
import se.tink.core.models.misc.Amount;

public interface RedirectionReceiver {

	void showTransactionDetails(String transactionId);

	void showTransactionListForIds(List<String> transactionIds);

	void showAccountDetails(String accountId, boolean topOnly);

	void showTransfer(String transferId);

	void showTransfer(String sourceAccountId, String destinationAccountId, Amount amount);

	void showCategory(boolean topOnly);

	void showBudget(String id, String periodStart);
}

File: commons/src/main/java/se/tink/android/redirection/RedirectionReceiver.kt
package se.tink.android.redirection

import com.tink.model.budget.BudgetFilter
import com.tink.model.budget.BudgetPeriodicity
import com.tink.model.misc.Amount

interface RedirectionReceiver {
    fun showTransactionDetails(transactionId: String)
    fun categorizeTransaction(transactionId: String)
    fun showTransactionListForIds(transactionIds: List<String>)
    fun showAccountDetails(accountId: String)
    fun showBudget(id: String, periodStart: String)
    fun createBudget(
        name: String? = null,
        amount: Amount? = null,
        filter: BudgetFilter? = null,
        periodicity: BudgetPeriodicity? = null
    )
}
